module/modmanager/manager.go

```diff
@@ -144,8 +151,19 @@ func (mgr *Manager) add(ctx context.Context, conf config.Module, conn *grpc.Clie
 		return nil
 	}
 
+	var moduleDataDir string
+	if mgr.moduleDataParentDirectory != "" {
```

Comment:
Documenting the intent here will be useful, such as:
* Under what circumstances is this not empty
* The check on 158 seems like a tautology. `moduleDataDir` is constructed as a join of `moduleDataParentDirectory`. Explaining what the expected environmental inputs are such that `Clean` can result in not finding a prefix would help future maintainers.